dummyfs/dummyfs.c

```diff
@@ -681,29 +681,18 @@ int dummyfs_read(void *ctx, oid_t *oid, offs_t offs, char *buff, size_t len)
 	dummyfs_object_t *o;
 
 	o = object_get(fs, oid->id);
-
-	if (o == NULL)
-		return -EINVAL;
-
-	if (!S_ISREG(o->mode) && !S_ISLNK(o->mode) && o->mode != OBJECT_MODE_MEM)
-		ret = -EINVAL;
-
-	if (buff == NULL)
-		ret = -EINVAL;
-
-	if (o->size <= offs) {
-		object_put(fs, o);
-		return 0;
+	if (o == NULL) {
+		return -ENOENT;
```

Comment:
I'd leave `EINVAL` here. `read()` operation doesn't define `ENOENT` as a valid errno.

Reply:
On the other hand such situation within read()/write() regime shouldn't be even possible - to access non-existing oid via existing fd ;) Anyway, I can change it back

Reply:
I agree, `o == NULL` shouldn't ever happen for a valid fd. Nonetheless I'd stick to `-EINVAL`.